### DISCUSSION TITLE
[help] What is the preferred way to get attributes of an individual target?

### DISCUSSION THREAD

**Discussion by billdenney** [ropensci/targets#1251] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'm working to get the `nlmixr2targets` package out to CRAN.  For that, I'm building tests on the targets that are generated.

I'm looking to extract the names and commands from the generated targets to test that they're generated correctly.  The best ways that I've found to get the name and command from the generated targets are:

* `generated_target$settings$name` gives the name as a character string.
* `generated_target$command$expr` gives the command to be run as an expression.

Are these the best ways to get the name and command, or is there a getter for target attributes?

**Comment by wlandau** [comment, marked as answer]
There isn't an official exported `targets` getter. It's generally better to control names and expressions yourself and pass them to `tar_target_raw()`, as opposed to relying on internal data in a target object, because the design of a target object might change in the future. However, if you absolutely must use e.g. `generated_target$settings$name`, you probably won't hit problems because I have not needed to change the structure of internal target data in the 4 years `targets` has been around.

**Reply by billdenney** [reply]
Thank you for this confirmation.

When using the objects, I'm passing the names, etc. around.  I was asking this for testing that the targets are created correctly (https://github.com/nlmixr2/nlmixr2targets/blob/main/tests/testthat/test-tar_nlmixr_multimodel.R).  In other words, I'm making sure that my package is interacting with `targets` correctly.  For that, I don't want to trust that the character string of a name that I'm passing around is correct but that the final target name is correct.

**Reply by wlandau** [reply]
Extra testing is a great idea. You could write unit tests with `tar_manifest()` and `tar_network()` to make sure the names, commands, and dependency graphs look right. I take this approach in `jagstargets` and `stantargets`.
